### DISCUSSION TITLE
Default Authentication Method SSO

### DISCUSSION THREAD

**Discussion by sdk1-kod** [osTicket/osTicket#6666] [Q&A]
Is it possible to configure osTicket with a default/force authentication method like for SSO?

**Comment by JediKev** [comment]
@thesorskod 

For Agents you can set their Authentication Backend to whatever backend you want and they will only be able to use that backend to login. Say you set it to OAuth2 SSO and the Agent attempts to login using local information, the system will simply deny them.

For Users you can set the Registration to Private and enable the setting to require login to create Tickets. This will remove the option for Users to register their own accounts so they have to either type in their local auth details or use SSO. If they use SSO the system will link their auth from the external provider to their local osTicket account via email/username. If they do not exist the system will auto-register an account for them (since SSO is considered pre-authorized users).

The caveat about this is our OAuth2 SSO plugin adds a "sign in with" button to the Client Portal and Agent Panel login pages. This means the local auth box will still appear. This can be a little confusing for some however they just need to learn to click the button to login with the external provider instead of directly typing their external credentials in the login box.

Cheers.

**Reply by RikkiMasters** [reply]
@JediKev is it possible to set all agents and default new ones to SSO? Or do they need to be done individually?
